packages/core/src/virtual-vendor.ts

```diff
@@ -0,0 +1,88 @@
+import { type Package } from '@embroider/shared-internals';
+import type { V2AddonPackage } from '@embroider/shared-internals/src/package';
+import { readFileSync } from 'fs';
+import { sortBy } from 'lodash';
+import resolve from 'resolve';
+import type { Engine } from './app-files';
+import type { Resolver } from './module-resolver';
+import type { VirtualContentResult } from './virtual-content';
+
+export function decodeVirtualVendor(filename: string): boolean {
+  return filename.endsWith('-embroider-vendor.js');
+}
+
+export function renderVendor(filename: string, resolver: Resolver): VirtualContentResult {
+  const owner = resolver.packageCache.ownerOfFile(filename);
+  if (!owner) {
+    throw new Error(`Failed to find a valid owner for ${filename}`);
+  }
+  return { src: getVendor(owner, resolver), watches: [] };
+}
+
+function getVendor(owner: Package, resolver: Resolver): string {
+  let engineConfig = resolver.owningEngine(owner);
+  let engine: Engine = {
+    package: owner,
+    addons: new Map(
+      engineConfig.activeAddons.map(addon => [
+        resolver.packageCache.get(addon.root) as V2AddonPackage,
+        addon.canResolveFromFile,
+      ])
+    ),
+    isApp: true,
+    modulePrefix: resolver.options.modulePrefix,
+    appRelativePath: 'NOT_USED_DELETE_ME',
+  };
+
+  // TODO - From where do we get this dynamically? in compat-app-builder:
+  // let emberENV = this.configTree.readConfig().EmberENV;
```

Comment:
note for when we're discussing this: all communication between stage 1 & 2 and stage 3 has to go through a file on disk in the `.embroider` folder. Right now we only have a `resolver.json` file but we might need to discuss having the config tree from addons be output on disk 👍